File: src/main/java/org/fountainmc/api/entity/data/hanging/HangingEntityData.java
package org.fountainmc.api.entity.data.hanging;

import javax.annotation.Nullable;

import org.fountainmc.api.Direction;
import org.fountainmc.api.entity.data.EntityData;

public interface HangingEntityData extends EntityData {

    /**
     * Get the Direction the Entity is facing
     * The Direction will be null for Entities like LeashKnots.
     *
     * @return the Direction the Entity is facing
     */
    @Nullable
    Direction getDirection();

    @Override
    default void copyDataFrom(EntityData data) {
        EntityData.super.copyDataFrom(data);
        if (data instanceof HangingEntityData) {
            // if we ever get anything to copy add it here
        }
    }
}

File: src/main/kotlin/org/fountainmc/api/entity/data/hanging/HangingEntityData.kt
package org.fountainmc.api.entity.data.hanging

import org.fountainmc.api.Direction
import org.fountainmc.api.entity.data.EntityData

interface HangingEntityData : EntityData {

    /**
     * The direction the Entity is facing
     *
     * The Direction will be null for Entities like LeashKnots.
     */
    val direction: Direction?

    /**
     * Copy all of the mutable properties from the given data into this object.
     *
     * The [direction] won't be copied, as it can't be sensibly modified for existing entities.
     */
    override fun copyDataFrom(data: EntityData) {
        super.copyDataFrom(data)
    }
}
